spec/emerald_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'spec_helper'
+require 'tempfile'
+require 'helper/user_comms'
+include Emerald
+require 'emerald/parser'
+require 'emerald/compiler'
+
```

Comment:
I disagree, I think that DRY has it's place everywhere, but should be exercised appropriately.
For example, I don't think that removing this line from the after hook and adding it to the tests adds any value, and if anything would contribute to further clutter of the tests:
{{code}}
 after(:each) do
      test_file.unlink
 end
{{code}}
Likewise, I can't see why I wouldn't extract this repeated code:
{{code}}ruby
test_file.write(lisp_expresssion)
test_file.close
{{code}}
into a separate, appropriately-named method which illustrates its purpose e.g. 
{{code}}ruby
def write_to_test_file(lisp_expresssion)
    test_file.write(lisp_expresssion)
    test_file.close
end
{{code}}
And call on it each time it's needed. I don't think that would reduce the clarity of the tests, but would reduce repetition. :)